### DISCUSSION TITLE
Hello everyone !!

### DISCUSSION THREAD

**Discussion by Elmatbaisaad** [aldas/modbus-tcp-client#110] [Q&A] [answered]
does anyone got  Call to undefined function ModbusTcpClient\Network\mb_strtolower() error 
here is my code for reading holding registers : 
![Capture d’écran 2022-04-13 165741](https://user-images.githubusercontent.com/101296742/163209601-4a8fa287-12b3-4459-a5e3-c2637fb6801d.png)

**Comment by aldas** [comment]
You need to have PHP `mbstring` extension installed

**Comment by Elmatbaisaad** [comment]
i tried but i failed , i m using docker with php fpm , please do you know how can I install the extension ?

**Reply by aldas** [reply]
maybe this helps https://stackoverflow.com/a/66980297/2514290

**Comment by Elmatbaisaad** [comment, marked as answer]
Thank i think it works because know im gettinh a other  error , this is the error o got :
[error.pdf](https://github.com/aldas/modbus-tcp-client/files/8486755/error.pdf)

**Reply by aldas** [reply]
your application tries to connect to 127.0.0.1:12345 and fails. I doubt your modbus server in running in that same container, on port 12345.  This is probably misconfigured URL

**Reply by Elmatbaisaad** [reply]
thank u for ur reply , i m using a python server (pymodbusTcp library)

**Reply by Elmatbaisaad** [reply]
I m really stuck in this error

**Reply by aldas** [reply]
you need to provide correct IP for that. Unless you run your container with ` --network="host"` your container has their "own" network stack/ip and needs to know IP where that  python server is running.  If this python modbus server is another Docker container you can use that container name instead of ip. 

Search for "docker network between containers" from Google.  ala https://www.tutorialworks.com/container-networking/ etc

**Reply by aldas** [reply]
`docker-compose` is a nice way to create multicontainer setups.

**Reply by Elmatbaisaad** [reply]
OK thank you i will try those suggestions.

**Reply by Elmatbaisaad** [reply]
The probleme here is that my server isn't in any container he is runing in my local machine

**Reply by aldas** [reply]
In that case instruct docker to use host network for that container.  See https://docs.docker.com/network/network-tutorial-host/
